lib/team_season_collection.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'team_season'
+require_relative 'season_collection'
+require_relative './modules/gatherable'
+require 'csv'
+
+class TeamSeasonCollection < Collection
+  include Gatherable
+
+  def initialize(csv_file_path)
+    super(csv_file_path, TeamSeason)
+  end
+
+  def home_id_parse(row, collection_type, home_id, hash)
+    season_hash = team_hash(row, home_id)
+    season_hash = team_season_hash(row, collection_type, season_hash, home_id)
+    season_data = season_data_array(season_hash, home_id)
+    key = team_key(season_hash)
+    season_key = season_key(season_hash, key)
+    hash = season_parse(key, season_key, season_data, hash)
+  end
+
+  def away_id_parse(row, collection_type, away_id, hash)
+    season_hash = team_hash(row, away_id)
+    season_hash = team_season_hash(row, collection_type, season_hash, away_id)
+    season_data = season_data_array(season_hash, away_id)
+    key = team_key(season_hash)
+    season_key = season_key(season_hash, key)
+    hash = season_parse(key, season_key, season_data, hash)
```

Comment:
In this code block the hash variable is inherited from the `.reduce` method, and it is set to a value with this method in order to start accumulating data. It is then reused in the next method to continue the process.

Reply:
Good explanation. I think, ultimately, that Hound is telling you this info because it doesn't really look too far ahead with where these variables may or may not be being used. Thanks for touching on this error.